### ISSUE TITLE
Is skipping ValueFormat1 == 0 in class kerning correct?

### ISSUE THREAD

**Issue by justvanrossum** [adobe-type-tools/kern-dump#1]
In line 250 of `getKerningPairsFromOTF.py` I see this:
```python
                            elif valueFormat == 0:
                                # valueFormat zero is caused by a value of <0 0 0 0> on a class-class pair; skip these
                                continue
```
Is this really correct? If valueFormat == 0, *no* kerning data from the entire subtable is read at all, which makes me wonder what the use of such a subtable is. (I'm looking at a font with a PairPosFormat2 subtable with ValueFormat1 == 0, and I can't quite figure out why it's there and how it should behave.)

**Comment by frankrolf** [comment]
I wonder if this represents a class-class pair with an explicit zero kerning value and one (or several) non-zero exceptions.

**Comment by behdad** [comment]
It's possible that ValueFormat2 is nonzero, which means second glyph in pairs is adjusted but no first glyph.

**Comment by justvanrossum** [comment]
That's not the case here, ValueFormat2 is also 0.

**Comment by behdad** [comment]
Then yes, that's a degenerate subtable.  Is Coverage empty as well?

**Comment by justvanrossum** [comment]
No, Coverage is not empty, and both ClassDefs are filled, too. It was compiled by AFDKO.

If I do _not_ ignore that subtable (but take those values as 0) I don't get the correct kerning out of the font, so that suggests that getKerningPairsFromOTF.py is doing the right thing. But then, why did AFDKO put it in there...

I may not pursue this further, though, as I have something that seems to works well now. This issue can be labeled "just curious" :)

**Comment by behdad** [comment]
Then that subtable will act as a stopper.  Anything on the coverage gets a zero there and next subtables will NOT be search for those glyphs as the first in a kern pair.

**Comment by justvanrossum** [comment]
Ok, but then there is after all something wrong with either the font or getKerningPairsFromOTF.py or both.

**Comment by behdad** [comment]
Not arguing with that :).

**Comment by frankrolf** [comment]
Just, would it be possible to send over the font? I am curious now!

**Comment by justvanrossum** [comment]
Emailed you.

**Comment by behdad** [comment]
I also like to see the font.  I read the code and found a few different issues.  Filing new issue now.

**Comment by behdad** [comment]
For reference: https://github.com/adobe-type-tools/kern-dump/issues/2
